client/src/models/opportunityfilterproxymodel.cpp

```diff
@@ -183,5 +183,21 @@ bool OpportunityFilterProxyModel::lessThan(const QModelIndex &left, const QModel
             return leftDt < rightDt;
         }
```

Comment:
Once the code below works, you could adjust the code above to benefit from it.
As in, delete the whole block above, and in the new fallback code, fallback to LastModifiedDate if sortColumn == nextStepDateColumn, and to CreationDate otherwise.